internal/storage/hub/bundle.go

```diff
@@ -236,31 +237,49 @@ func (b *Bundle) loadPolicySet(idHex, fileName string) (*runtimev1.RunnablePolic
 }
 
 func (b *Bundle) InspectPolicies(ctx context.Context, listParams storage.ListPolicyIDsParams) (map[string]*responsev1.InspectPoliciesResponse_Result, error) {
-	policyIDs, err := b.ListPolicyIDs(ctx, listParams)
+	filteredFQNs, err := b.ListPolicyIDs(ctx, listParams)
 	if err != nil {
 		return nil, fmt.Errorf("failed to list policies: %w", err)
 	}
 
-	if len(policyIDs) == 0 {
+	if len(filteredFQNs) == 0 {
 		return nil, nil
 	}
 
-	results := make(map[string]*responsev1.InspectPoliciesResponse_Result)
-	for _, policyID := range policyIDs {
-		id := namer.GenModuleIDFromFQN(policyID)
+	fqns, err := b.ListPolicyIDs(ctx, storage.ListPolicyIDsParams{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list policies without list parameters: %w", err)
+	}
+
+	ins := inspect.PolicySets()
+	for _, fqn := range fqns {
+		id := namer.GenModuleIDFromFQN(fqn)
 		idHex := id.HexStr()
 		fileName := policyDir + idHex
 
 		pset, err := b.loadPolicySet(idHex, fileName)
 		if err != nil {
-			return nil, fmt.Errorf("failed to load policy %s: %w", policyID, err)
+			return nil, fmt.Errorf("failed to load policy %s: %w", fqn, err)
 		}
 
-		actions := policy.ListPolicySetActions(pset)
-		if len(actions) > 0 {
-			results[namer.PolicyKeyFromFQN(pset.Fqn)] = &responsev1.InspectPoliciesResponse_Result{
-				Actions: actions,
-			}
+		if err = ins.Inspect(pset); err != nil {
+			return nil, fmt.Errorf("failed to inspect policy %s: %w", fqn, err)
```

Comment:
Isn't this inspecting the whole bundle and not just the ones the user asked for?